src/saltext/vmware/states/vmc_nat_rules.py

```diff
@@ -0,0 +1,512 @@
+        )
+
+    if __opts__.get("test"):
+        log.info("present is called with test option")
+        if existing_nat_rule:
+            return vmc_state._create_state_response(
+                name=name, comment="State present will update nat rule {}".format(nat_rule)
+            )
+        else:
+            return vmc_state._create_state_response(
+                name=name, comment="State present will create nat rule {}".format(nat_rule)
+            )
+
+    if existing_nat_rule:
+        updatable_keys = input_dict.keys()
+        is_update_required = vmc_state._check_for_updates(
+            existing_nat_rule, input_dict, updatable_keys, ["translated_ports", "tags"]
+        )
+
+        if is_update_required:
+            updated_nat_rule = __salt__["vmc_nat_rules.update"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                tier1=tier1,
+                nat=nat,
+                nat_rule=nat_rule,
+                verify_ssl=verify_ssl,
+                cert=cert,
+                action=action,
+                destination_network=destination_network,
+                source_network=source_network,
+                translated_network=translated_network,
+                translated_ports=translated_ports,
+                scope=scope,
+                service=service,
+                enabled=enabled,
+                firewall_match=firewall_match,
+                logging=logging,
+                description=description,
+                tags=tags,
+                sequence_number=sequence_number,
+                display_name=display_name,
+            )
+
+            if "error" in updated_nat_rule:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_nat_rule["error"], result=False
+                )
+
+            get_nat_rule_after_update = __salt__["vmc_nat_rules.get_by_id"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                tier1=tier1,
+                nat=nat,
+                nat_rule=nat_rule,
+                verify_ssl=verify_ssl,
+                cert=cert,
+            )
+
+            if "error" in get_nat_rule_after_update:
+                return vmc_state._create_state_response(
+                    name=name, comment=get_nat_rule_after_update["error"], result=False
+                )
+
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Updated nat rule {}".format(nat_rule),
+                old_state=existing_nat_rule,
+                new_state=get_nat_rule_after_update,
+                result=True,
+            )
```

Comment:
**suggestion** This behavior could be delegated to `_create_state_response`, since `error` in new_state appears to always trigger `result=False`. It would be possible to update the response to override it like so:

```
result = 'error' in new_state if new_state and result is None else result
```

Of course, a result of `None` *does* have meaning, so if there is a case where the above approach doesn't work, the response function signature could be written like:

```
MISSING = object()

def _create_state_response(..., result=MISSING):
    ...
    result = 'error' in new_state if new_state and result is MISSING else result
```

Reply:
I looked at it, we have comment field which changes based on failure and success. So keeping it as it is, making no change